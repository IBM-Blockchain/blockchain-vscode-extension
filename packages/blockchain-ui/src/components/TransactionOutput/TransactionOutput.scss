@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/layout/scss/_spacing';

.output-panel {
    position: fixed;
    height: 100%;
    width: 52%;
    background-color: #F4F4F4;
    padding: $spacing-07;

    .output-title {
        font-size: 0.875rem;
        color: #393939;
    }

    .output-body {
        @include carbon--font-face-mono;
        font-family: 'IBM Plex Mono';
        font-size: 0.875rem;
        color: inherit;
    }
}
