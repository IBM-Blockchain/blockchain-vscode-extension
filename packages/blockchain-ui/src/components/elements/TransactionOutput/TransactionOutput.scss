@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/layout/scss/_spacing';

.output-panel {
    position: fixed;
    top: 0px;
    height: 100%;
    width: 55%;
    background-color: #F4F4F4;

    .output-panel-inner {
        margin: $spacing-07;
        padding-right: $spacing-07 + $spacing-09;


        .output-title {
            font-size: 0.875rem;
            color: #393939;
        }

        .output-body {
            @include carbon--font-face-mono;
            font-family: 'IBM Plex Mono', 'IBM Plex Sans', 'IBMPlexSans', Arial, Helvetica, sans-serif;
            font-size: 0.875rem;
            color: inherit;
        }

    }
}
