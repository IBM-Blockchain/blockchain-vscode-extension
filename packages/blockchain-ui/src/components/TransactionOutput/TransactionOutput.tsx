import React, { Component } from 'react';
import './TransactionOutput.scss';

interface OutputProps {
    output: string;
}

interface OutputState {
    output: string;
}

class TransactionOutput extends Component<OutputProps, OutputState> {
    constructor(props: Readonly<OutputProps>) {
        super(props);
        this.state = {
            output: this.props.output
        };
        this.determineOutput = this.determineOutput.bind(this);
    }

    componentDidUpdate(prevProps: OutputProps): void {
        if (prevProps.output !== this.props.output) {
            this.setState({
                output: this.props.output
            });
        }
    }

    determineOutput(): Array<JSX.Element>  {
        const outputArray: Array<JSX.Element> = [];
        if (this.state.output === '') {
            outputArray.push(<p className='output-body'>No transaction output available. Submit/evaluate to produce an output.</p>);
        } else {
            const outputStrings: Array<string> = this.state.output.split('\n');
            for (const output of outputStrings) {
                outputArray.push(<p className='output-body'>{output}</p>);
            }
        }
        return outputArray;
    }

    render(): JSX.Element {
        return (
            <div className='output-panel' id='output-panel'>
                <p className='output-title'>Output for untitled transaction</p>
                {this.determineOutput()}
            </div>
        );
    }
}

export default TransactionOutput;
