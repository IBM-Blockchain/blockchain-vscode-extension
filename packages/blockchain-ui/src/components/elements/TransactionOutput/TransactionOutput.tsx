import React, { FunctionComponent } from 'react';
import { Launch32 } from '@carbon/icons-react';
import { ExtensionCommands } from '../../../ExtensionCommands';
import CommandLink from '../CommandLink/CommandLink';
import './TransactionOutput.scss';
import block from '../../../resources/block.svg';

interface IProps {
    output: string;
}

const TransactionOutput: FunctionComponent<IProps> = ({ output }) => {
    return (
        <div className='output-panel' id='output-panel'>
            <p className='output-title'>Transaction output</p>
            <div className='output-panel-inner'>
                {output
                    ?  output.split('\n').map((line) => <p className='output-body'>{line}</p>)
                    : (
                        <div className='output-placeholder-container'>
                            <div className='output-placeholder'>
                                <img src={block} alt='block icon' />
                                <h4>No transaction output, yet!</h4>
                                <p>Submit or evaluate a transaction to view it's output here.</p>
                                <CommandLink
                                    linkContents={<>Learn more <Launch32/></>}
                                    commandName={ExtensionCommands.OPEN_TUTORIAL_PAGE}
                                    commandData={['Basic tutorials', 'A4: Invoking a smart contract from VS Code']}
                                />
                            </div>
                        </div>
                    )
                }
            </div>
        </div>
    );
};

export default TransactionOutput;
