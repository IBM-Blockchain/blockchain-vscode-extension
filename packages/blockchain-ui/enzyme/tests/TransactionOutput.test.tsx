// tslint:disable no-unused-expression
import React from 'react';
import renderer from 'react-test-renderer';
import TransactionOutput from '../../src/components/transactionView/TransactionOutput/TransactionOutput';
import chai from 'chai';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('TransactionOutput component', () => {
    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<TransactionOutput output='some output'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the expected snapshot when no output is received', async () => {
        const component: any = renderer
            .create(<TransactionOutput output=''/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
