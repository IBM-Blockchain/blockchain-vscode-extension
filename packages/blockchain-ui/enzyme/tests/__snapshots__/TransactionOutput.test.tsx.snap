// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionOutput component should render the expected snapshot 1`] = `
<div
  className="output-panel"
  id="output-panel"
>
  <p
    className="output-title"
  >
    Output for untitled transaction
  </p>
  <p
    className="output-body"
  >
    some output
  </p>
</div>
`;
